### PR TITLE
[Fix] fix only cuda device bug in datasets sampler

### PR DESCRIPTION
**Pull request by jessiewiswjc**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

The code raise error when i am running mmdet3d pgd network on MLU device:

image

## Modification

Add utils function to get current device and pass it to `sync_random_seed` function.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/datasets/samplers/infinite_sampler.py [hunk 1]
```diff
@@ -56,7 +57,8 @@ def __init__(self,
         # in the same order based on the same seed. Then different ranks
         # could use different indices to select non-overlapped data from the
         # same data list.
-        self.seed = sync_random_seed(seed)
+        device = get_device()
```
Comment (RangiLyu) [inline mmdet/datasets/samplers/infinite_sampler.py:60]: The `InfiniteBatchSampler` in this file has the same problem. Would you like to fix it in this PR?

Comment (jessiewiswjc) [reply mmdet/datasets/samplers/infinite_sampler.py:60]: Ok I have fixed it.

**Review by RangiLyu — APPROVED** [review]
LGTM

**Comment by vansin** [comment]
@jessiewiswjc 
Congratulations on having your first PR merged in OpenMMLab! I am a technical operations staff member of OpenMMLab. My WeChat ID is van-sin. I would like to invite you to join the OpenMMLab contributors WeChat group.

**Pull request merged**